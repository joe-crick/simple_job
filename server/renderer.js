import React from "react";
import { renderToString } from "react-dom/server";
import Main from "../src/index";
import { jobData } from "../src/data/job-test-data";
import StaticRouter from "react-router-dom/StaticRouter";
import { Provider } from "react-redux";
import fs from "fs";
import path from "path";
import rootStore from "../src/stores/store";
import appConfig from "../src/app.config";
import defaultState from "../src/state/state";

const context = {};
const filePath = path.resolve(__dirname, "..", "build", "index.html");
const template = fs.readFileSync(filePath, "utf8");

export default (req, res) => {
  defaultState.jobsList = jobData(appConfig.defaultCount);
  const appHtml = renderToString(
    <Provider store={rootStore(defaultState)}>
      <StaticRouter location={req.url} context={context}>
        <Main />
      </StaticRouter>
    </Provider>
  );

  return context.url
    ? res.redirect(301, context.url)
    : res.send(
        template
          .replace("{{SSR}}", appHtml)
          .replace("{{json}}", JSON.stringify(defaultState.jobsList))
      );
};
