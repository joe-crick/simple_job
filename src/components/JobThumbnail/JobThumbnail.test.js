import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import JobThumbnail from "./JobThumbnail";

describe("Job", () => {
  it("should render", () => {
    const component = renderer.create(<JobThumbnail />);
    let header = component.toJSON();
    expect(header).toMatchSnapshot();
  });
  it("should run a function when the buy button is clicked.", () => {
    const noop = jest.fn();
    const component = mount(<JobThumbnail buyNow={noop} />);
    component.find(".JobThumbnail-button").simulate("click");
    expect(noop).toHaveBeenCalled();
  });
});
