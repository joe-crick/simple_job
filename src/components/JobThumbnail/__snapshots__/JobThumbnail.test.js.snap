// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job should render 1`] = `
<div
  className="JobThumbnail undefined"
>
  <div
    className="JobThumbnail-logo"
  >
    <img
      alt={undefined}
      src={undefined}
    />
  </div>
  <div
    className="JobThumbnail-details"
  >
    <div
      className="JobThumbnail-price"
    >
      Price: â‚¬
    </div>
    <div
      className="JobThumbnail-title"
    />
    <div
      className="JobThumbnail-desc"
    />
    <button
      className="JobThumbnail-button"
      data-id={undefined}
      onClick={undefined}
    >
      Buy
    </button>
  </div>
</div>
`;
